frontend/src/pages/signup/SignUp.jsx

```diff
@@ -0,0 +1,225 @@
+  const [email, setEmail] = useState('');
+  const [verificationCode, setVerificationCode] = useState('');
+  const [verificationCodeCheck, setverificationCodeCheck] = useState(false);
+  const [inputCode, setInputCode] = useState('');
+  const [isCodeSent, setIsCodeSent] = useState(false);
+  const [timeLeft, setTimeLeft] = useState(0);
+  const [passwordInput, setpasswordInput] = useState(false);
+  const [passwordInputNum, setpasswordInputNum] = useState('');
+  const [passwordCheckInput, setpasswordCheckInput] = useState(false);
+  const form = useRef();
+
+  // 6자리 랜덤 코드 생성
+  const generateVerificationCode = () => {
+    return Math.random().toString(36).substring(2, 8).toUpperCase(); // 영어 + 숫자 조합 (6자리)
+  };
+
+  const emailCheck = (e) => {
+    e.preventDefault();
+
+    let email_format = /^([0-9a-zA-Z_\.-]+)@([0-9a-zA-Z_-]+)(\.[0-9a-zA-Z_-]+){1,2}$/;
+    if (!email || !email_format.test(email)) {
+      alert('이메일 형식이 올바르지 않습니다.');
+      return;
+    }
+
+    const code = generateVerificationCode();
+    setVerificationCode(code);
+
+    const templateParams = {
+      to_name: email,
+      message: code,
+      from_name: 'timepaper',
+    };
+
+    emailjs.send('service_4n5mcry', 'template_j25i3xn', templateParams, 'E2uVL_UZRQC6xI7Nu').then(
+      () => {
+        console.log('인증 코드 전송 성공!');
+        setIsCodeSent(true);
+        alert('인증 코드가 이메일로 전송되었습니다.');
+        setIsCodeSent(true);
+        setTimeLeft(300);
+      },
+      (error) => {
+        console.log('이메일 전송 실패...', error.text);
+        alert('이메일 전송에 실패했습니다.');
+      },
+    );
+  };
+
+  useEffect(() => {
+    if (timeLeft > 0) {
+      const timer = setTimeout(() => setTimeLeft(timeLeft - 1), 1000);
+      return () => clearTimeout(timer);
+    } else {
+      setverificationCodeCheck(false);
+    }
+  }, [timeLeft]);
+
+  const handleFocus = (index) => {
+    const newPlaceholdersVisible = [...placeholdersVisible];
+    newPlaceholdersVisible[index] = false;
+    setPlaceholdersVisible(newPlaceholdersVisible);
+  };
+
+  const handleBlur = (index, value) => {
+    const newPlaceholdersVisible = [...placeholdersVisible];
+    if (value === '') {
+      newPlaceholdersVisible[index] = true;
+    }
+    setPlaceholdersVisible(newPlaceholdersVisible);
+  };
+
+  const ChengeverificationCode = (e) => {
+    const value = e.target.value.replace(/(\s*)/g, '');
+    setInputCode(value);
+    if (value === verificationCode) {
+      setverificationCodeCheck(true);
+      console.log('성공');
+    } else {
+      setverificationCodeCheck(false);
+    }
+  };
+
+  const ChengePassword = (e) => {
+    const value = e.target.value.replace(/(\s*)/g, '');
+    setpasswordInputNum(value);
+
+    const check = /[`~!@#$%^&*|\\\'\";:\/?]/gi;
+
+    if (check.test(value) && value.length >= 8) {
+      setpasswordInput(true);
+    } else {
+      setpasswordInput(false);
+    }
+  };
+
+  const ChengePasswordCheck = (e) => {
```

Comment:
```suggestion
  const handleVerificationCode = (e) => {
    const value = e.target.value.replace(/(\s*)/g, '');
    setInputCode(value);
    if (value === verificationCode) {
      setverificationCodeCheck(true);
      console.log('성공');
    } else {
      setverificationCodeCheck(false);
    }
  };

  const handlePassword = (e) => {
    const value = e.target.value.replace(/(\s*)/g, '');
    setpasswordInputNum(value);

    const check = /[`~!@#$%^&*|\\\'\";:\/?]/gi;

    if (check.test(value) && value.length >= 8) {
      setpasswordInput(true);
    } else {
      setpasswordInput(false);
    }
  };

  const handlePasswordCheck = (e) => {
```
가독성을 향상을 위해
chenge로 쓰신 것 handle 로 변경부탁드리고 
카멜 케이스 적용 부탁드려요.